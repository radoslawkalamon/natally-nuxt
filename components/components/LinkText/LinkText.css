.link-text {
  color: inherit;
  text-decoration: underline;
}

.link-text:hover {
  text-decoration: none;
}
