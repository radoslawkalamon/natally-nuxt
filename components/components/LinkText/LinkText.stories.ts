import type { Meta, Story } from '@storybook/vue'
import ComponentsLinkText from '@/components/components/LinkText/LinkText.vue'

const meta: Meta = {
  title: 'Components / Link Text',
  component: ComponentsLinkText
}
export default meta

const Template: Story = (_args, { argTypes }) => ({
  components: {
    ComponentsLinkText
  },
  props: Object.keys(argTypes),
  template: '<ComponentsLinkText v-bind="$props" />'
})

export const Default: Story = Template.bind({})
Default.args = {
  to: '/the-url/',
  label: 'Item label'
}
