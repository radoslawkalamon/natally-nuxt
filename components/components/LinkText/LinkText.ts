import Vue from 'vue'

export default Vue.extend({
  name: 'ComponentsLinkText',
  inheritAttrs: false,
  props: {
    label: {
      type: String,
      required: true
    }
  }
})
