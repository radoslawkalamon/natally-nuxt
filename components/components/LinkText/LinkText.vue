<template>
  <NuxtLink
    class="link-text"
    v-bind="$attrs"
    v-on="$listeners"
    v-text="label"
  />
</template>

<script src="./LinkText.ts" lang="ts"></script>
<style src="./LinkText.css" scoped></style>
