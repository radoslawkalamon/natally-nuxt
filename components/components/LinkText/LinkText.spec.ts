import LinkText from '@/components/components/LinkText/LinkText.vue'
import { shallPassUnitSanityTest } from '@/devtools/jest.common.spec'

describe('Components / Link Text', () => {
  describe('Unit', () => {
    shallPassUnitSanityTest({
      component: LinkText,
      options: {
        propsData: {
          label: 'Item label'
        }
      }
    })
  })
})
