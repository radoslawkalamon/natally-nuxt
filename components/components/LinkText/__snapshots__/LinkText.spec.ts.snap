// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components / Link Text Unit shall pass unit sanity test 1`] = `"<nuxtlink-stub class=\\"link-text\\">Item label</nuxtlink-stub>"`;
